### PR TITLE
fix: wc "no matching key" errors

### LINKED ISSUE #3284: WalletConnect: no matching session key
## Bug description

* Connect a Safe to another Safe that the first one owns via WC
* Wait... (it can be slow – investigate)
* Once Approve/Rejects appears, press Approve
* Instead of connecting, it shows a "no matching session key" error

This is somehow related to timing and doesn't happen if you wait a bit before pressing Approve.

### PR DESCRIPTION
**Pull request by compojoom**
## What it solves

Resolves #3284

This is an attempt to fix 2 WC bugs that are plaguing some of our users:

No matching key: session
No matching key: keychain

For the keychain error I haven't figured a way how to reproduce it, but it appears that WC has fix in their latest release. For the session error I could reproduce it when connecting to 1inch, but also very edge case and I doubt that users are encountering it that way. We have a lot of errors in sentry and I can't imagine that users trigger it like this.

## How to test it
For the session error you could try the PR and follow the same steps as in the gif. The problem is that the fix is flaky - sometimes it works, sometimes it doesn't, where without it, it always fails. 

For the keychain issue, no idea. We'll have to monitor sentry for that error.

### CODE REVIEW

File: src/features/walletconnect/components/WcSessionMananger/index.tsx [hunk 1]
```diff
@@ -25,11 +25,14 @@ const WcSessionManager = ({ sessions, uri }: WcSessionManagerProps) => {
   const { walletConnect, error, setError, open, setOpen } = useContext(WalletConnectContext)
   const [proposal, setProposal] = useState<Web3WalletTypes.SessionProposal>()
   const [changedSession, setChangedSession] = useState<SessionTypes.Struct>()
+  const isResponded = useRef<boolean>(false)
```
Comment (katspaugh) [inline src/features/walletconnect/components/WcSessionMananger/index.tsx:28]: Do we need to reset this back to false when the proposal changs? Or is the entire component re-rendering then?

File: src/features/walletconnect/components/WcSessionMananger/index.tsx [hunk 2]
```diff
@@ -62,6 +66,21 @@ const WcSessionManager = ({ sessions, uri }: WcSessionManagerProps) => {
...
+  // but for whatever reason refreshes the browser without rejecting/accepting it
+  // if we don't do this we often get "No matching key: session"
+  useEffect(() => {
+    const handleBeforeUnload = async () => {
+      if (proposal && isResponded.current === false) {
+        await onReject()
+      }
+    }
+
+    window.addEventListener('beforeunload', handleBeforeUnload)
```
Comment (katspaugh) [inline src/features/walletconnect/components/WcSessionMananger/index.tsx:79]: AFAIR beforeunload is supposed to return a boolean to prevent the browser from closing? Maybe we should use onunload?

**Review by katspaugh — COMMENTED** [review]
Please adjust the title of the PR as per our convention.

File: src/features/walletconnect/components/WcSessionMananger/index.tsx (hunk 2 above)
Comment (katspaugh) [reply src/features/walletconnect/components/WcSessionMananger/index.tsx:79]: Thanks!

File: src/features/walletconnect/components/WcSessionMananger/index.tsx (hunk 1 above)
Comment (compojoom) [reply src/features/walletconnect/components/WcSessionMananger/index.tsx:28]: Good point! Fixed.

**Review by katspaugh — APPROVED** [review]

File: src/features/walletconnect/components/WcSessionMananger/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/features/walletconnect/components/WcSessionMananger/index.tsx:28]: Thanks!

Btw please don't resolve threads unless you started them so that the author can come back and see your response (and hopefully resolve it) 😄
